pypdf/_page_labels.py

```diff
@@ -173,3 +183,78 @@ def index2label(reader: PdfReaderProtocol, index: int) -> str:
         )
     # TODO: Implement /Kids and /Limits for number tree
     return str(index + 1)  # Fallback
+
+
+def nums_insert(
+    key: NumberObject,
+    value: DictionaryObject,
+    nums: ArrayObject,
+) -> None:
+    """
+    See 7.9.7 "Number Trees".
+    Insert a key, value pair in a Nums array.
+
+    Args:
+        key: number key of the entry
+        value: value of the entry
+        nums: Nums array to modify
+    """
+    if len(nums) % 2 != 0:
+        raise ValueError("a nums like array must have an even number of elements")
```

Comment:
Wether codecov is enogh: the only things left out are the exceptions of the nums functions.

Reply:
Don't worry too much about code coverage. It's nice if it's complete and it makes it a lot easier for me to review, but I will also accept PRs without tests as long as I'm convinced that it overall improves pypdf.